GVFS/GVFS.Common/Git/CredentialManager.cs

```diff
@@ -0,0 +1,29 @@
+﻿using GVFS.Common.Tracing;
+
+namespace GVFS.Common.Git
+{
+    public class CredentialManager : ICredentialManager
+    {
+        private GitProcess gitProcess;
+
+        public CredentialManager(GitProcess gitProcess)

```

Comment:
Since all the methods of this class take an `ITracer`, I would move that to the constructor, that will most likely cause other classes like `GitAuthentication` to take it in the constructor as well.

Reply:
This is actually what I attempted initially. The problem that I ran into was that not only does `GitAuthentication` need to take an `ITracer`, but also the classes that build `GitAuthentication`, such as `Enlistment`, and `GVFSEnlistment`, all the way to `GVFSVerb` ([here](https://github.com/microsoft/VFSForGit/blob/master/GVFS/GVFS/CommandLine/GVFSVerb.cs#L844))

Reply:
This change has been scoped down a bit - instead of a new `CredentialStore` class, the functionality will be implemented in `GitProcess`